### PR TITLE
Test combine expt xfail

### PR DESCRIPTION
**Pull request by graeme-winter**
For #1093 debugging

### CODE REVIEW

File: tests/command_line/test_combine_experiments.py [hunk 1]
```diff
@@ -423,3 +423,30 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
     assert "Beam" in str(exc.value)
     print("Got (expected) error message:", exc.value)
+
+
+@pytest.mark.xfail(
+    strict=True,
+    raises=AssertionError,
+    reason="https://github.com/dials/dials/issues/1093",
+)
+def test_combine_imagesets(dials_data, tmpdir):
```
Comment (benjaminhwilliams) [inline tests/command_line/test_combine_experiments.py:433]:
```suggestion
def test_combine_imagesets(dials_data, tmp_path):
```

File: tests/command_line/test_combine_experiments.py [hunk 2]
```diff
@@ -423,3 +423,30 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
+    data = dials_data("l_cysteine_dials_output")
+
+    for command in (
+        f"dials.import {data}/l-cyst_01_000*.cbf output.experiments=sweep1.expt",
+        f"dials.import {data}/l-cyst_02_000*.cbf output.experiments=sweep2.expt",
```
Comment (benjaminhwilliams) [inline tests/command_line/test_combine_experiments.py:438]: I think this makes for a better OS-agnostic test.  Happy to be overruled though.
```suggestion
        f"dials.import {data / 'l-cyst_01_000*.cbf'} output.experiments=sweep1.expt",
        f"dials.import {data / 'l-cyst_02_000*.cbf'} output.experiments=sweep2.expt",
```

File: tests/command_line/test_combine_experiments.py [hunk 3]
```diff
@@ -423,3 +423,30 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
+        "dials.combine_experiments index1.expt index1.refl index2.expt index2.refl",
+    ):
+        result = procrunner.run(command.split(), working_directory=tmpdir)
+        assert not result.returncode and not result.stderr
+
+    comb = flex.reflection_table.from_file(os.path.join(tmpdir, "combined.refl"))
+
+    iset = comb["imageset_id"]
+
+    assert flex.max(iset) == 1
```
Comment (benjaminhwilliams) [inline tests/command_line/test_combine_experiments.py:452]: Might as well check that we have two imageset IDs, while we're at it?
```suggestion
    assert set(iset) == {0, 1}
```

File: tests/command_line/test_combine_experiments.py [hunk 4]
```diff
@@ -423,3 +423,30 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
+    for command in (
+        f"dials.import {data / 'l-cyst_01_000*.cbf'} output.experiments=sweep1.expt",
+        f"dials.import {data / 'l-cyst_02_000*.cbf'} output.experiments=sweep2.expt",
+        "dials.find_spots sweep1.expt output.reflections=strong1.refl",
+        "dials.find_spots sweep2.expt output.reflections=strong2.refl",
+        "dials.index sweep1.expt strong1.refl output.reflections=index1.refl output.experiments=index1.expt",
+        "dials.index sweep2.expt strong2.refl output.reflections=index2.refl output.experiments=index2.expt",
+        "dials.combine_experiments index1.expt index1.refl index2.expt index2.refl",
+    ):
+        result = procrunner.run(command.split(), working_directory=tmpdir)
```
Comment (benjaminhwilliams) [inline tests/command_line/test_combine_experiments.py:445]: Oops, sorry, missed this one.
```suggestion
        result = procrunner.run(command.split(), working_directory=tmp_path)
```

File: tests/command_line/test_combine_experiments.py [hunk 5]
```diff
@@ -423,3 +423,30 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
+        assert not result.returncode and not result.stderr
+
+    comb = flex.reflection_table.from_file(os.path.join(tmpdir, "combined.refl"))
```
Comment (benjaminhwilliams) [inline tests/command_line/test_combine_experiments.py:448]: And this one.
```suggestion
    comb = flex.reflection_table.from_file(os.path.join(tmp_path, "combined.refl"))
```

**Review by benjaminhwilliams — CHANGES_REQUESTED** [review]

**Review by benjaminhwilliams — APPROVED** [review]

**Comment by ndevenish** [comment]
This seems to test the same thing without spending a couple of minutes burning energy

**Comment by benjaminhwilliams** [comment]
Looks good.

File: tests/command_line/test_combine_experiments.py [hunk 6]
```diff
@@ -423,3 +423,22 @@ def test_failed_tolerance_error(dials_regression, monkeypatch):
...
+
+@pytest.mark.xfail(
+    strict=True,
+    raises=AssertionError,
+    reason="https://github.com/dials/dials/issues/1093",
+)
+def test_combine_imagesets(dials_data, tmp_path):
+    data = dials_data("vmxi_proteinase_k_sweeps", pathlib=True)
+
+    os.chdir(tmp_path)
```
Comment (Anthchirp) [inline tests/command_line/test_combine_experiments.py:436]: Must ensure CWD is not changed beyond the lifetime of test

**Review by Anthchirp — CHANGES_REQUESTED** [review]

**Review by Anthchirp — APPROVED** [review]
My recollection of the chdir issue was flawed 👍

**Pull request merged**